tests/test.py

```diff
@@ -9714,6 +9715,43 @@ def testClientIDRead(self):
         self.assertEqual(ex.exception.errno, MYSQL_ERR_QUERY_INTERRUPTED_CODE)
 
 
+class TestMySQLTextCharsetLiterals(TestMySQLTextTypeAwareDecryptionWithoutDefaults):
+    def testClientIDRead(self):
+        """
+        We don't directly support charset introducers[1], but at least make
+        sure it can handle _binary.
+
+        [1]: https://dev.mysql.com/doc/refman/8.0/en/charset-introducer.html
+        """
+
+        self.schema_table.create(bind=self.engine_raw, checkfirst=True)
+        id = get_random_id()
+        insert = self.test_table.insert().values(
```

Comment:
hm, if sqlalchemy compile query with parentheses, then lets test them both? and use raw sql query to use without parentheses if sqlalchemy doesn't allow compile without them